### ISSUE TITLE
Delete a button delete all events

### ISSUE THREAD

**Issue by jonathanathe** [blakeblackshear/frigate#2223]
**Describe what you are trying to accomplish and why in non technical terms**
I think it is necessary to add a button in which by pressing it you can delete all the recorded events and images.It happens to me that many recordings are triggered and it causes my home Assistant to fill up its memory with 60gb. So in a quick way after reviewing the events to be able to eliminate them in one go instead of having to go one by one to eliminate it.

**Describe the solution you'd like**
Have a button "Delete all events" that when pressed I deleted all the recordings or photos made.

**Comment by hellboy-wien** [comment]
Same here, today i has to stop recording on motion... bc i cant delete every event manualy

**Comment by jayjayseal** [comment]
I can relate... full disk and no easy way to delete events...

I can however try and delete just the content of the folder but there might be some database behind it that does not get wiped?
I think i need to do this anyhow or the host get's all kinds of issues

**Comment by jayjayseal** [comment]
Indeed, the frigate database does not get cleaned when deleting the images and video's. i still have to manually delete each individual event. i have set the retention to 1 day but it does not run on booting of frigate and it did not run during the night. I'll wait a bit longer today and see if i might have missed the run.

Oh well.... my free space jumped back to 30% instead of nearly full :+1:

**Comment by ruant** [comment]
Maybe it should be able to select multiple events and delete them instead.
Would cover more options people might have.
Events could be sorted inside the table view by dates, so by checking the checkbox for the day label, it selects all events that day.
Then you could delete them.
Also then have a checkbox all the way up top that selects EVERYTHING, and you can do a delete with that if you want to delete every single event.

**Comment by rainman369** [comment]
Bumping this - recording trigger without me knowing yesterday so I have a buttload of clips currently that I really do not feel like deleting one by one

**Comment by jayjayseal** [comment]
This should be implemented. the current way of working is terrible. Please include this functionality. This can't be hard to implement and would be such a good feature.

**Comment by jayjayseal** [comment]
Let me begin with my express my sincere apologies to the devs.... I found out what i did wrong and now frigate is behaving like it supposed to do when auto purging events etc.

This will probably solve most of the problems for the users in this same thread.

For me it was even a 2 part error and your error might be similar. 

The first mistake i made:

I think i might have missed the memo about the changes to the config file.

**Expire_days** was replaced by just **days** in the config file:

This is a working config:
```
record:
  enabled: True
  expire_interval: 600
  retain:
    days: 3
    mode: all
  events:
    retain:
      default: 3
      mode: all
```
If you adjust this in the right config file it should work, and that is the next error i made.

I'm running HomeAssistant with the **frigate proxy add-on** and since i'm not running frigate on the same machine as HomeAssistant... adjusting or creating the config.yml in the root directory of HomeAssistant along with the configuration of homeassistant itself... is... kind of pointless :-) This is however the option that most forms etc would recommend as they run the normal Frigate server along with homeassistant (i guess)

What you in this case should then do is adjust the config.yml on the machine running frigate. in my case i had to login to the docker container and adjust the config from there (or mount the frigate folder on the host running frigate to your device and adjust if from there).

I found this out because the logs of frigate kept complaining about deprecated values in the config... but the error remained after i changed the config file.... i could empty the .yml file and it still would give me the same error.

_Though... mass deletion of events would be a much appreciated add-in for this awesome piece of software. (because it is! <3 )_

**Comment by MrAlucardDante** [comment]
Bump, we could always use sql to delete all the events, but we should have to manually update the database to performs basic tasks

**Comment by doubleopinter** [comment]
Bump again, I'd like to see some way of bulk deleting events as well.

**Comment by jayjayseal** [comment]
> Bump again, I'd like to see some way of bulk deleting events as well.

Yes, this is a much missed option..

**Comment by bd94** [comment]
> > Bump again, I'd like to see some way of bulk deleting events as well.
> 
> Yes, this is a much missed option..

Bumping once again, have had to delete events manually a few times now, one at a time. Would love an option to bulk select events to delete.

**Comment by Flaxee** [comment]
I'm also missing this option to delete multiple/all events at once.

**Comment by NickM-27** [comment]
The [beta for the latest 0.11 release](https://github.com/blakeblackshear/frigate/releases/tag/v0.11.0-beta2) now allows deleting of a single click which is equivalent to selecting and deleting multiple events at a time (at least in terms of user clicks required). Check it out 👍

**Comment by jayjayseal** [comment]
Then it's our job to test this out :-) i'm jumping in! tnx for notifying!

**Comment by bd94** [comment]
I'm running the latest beta2 docker image. Much better and much less tedious deleting events to clean up space if needed.
On Firefox though the delete icon is massive and takes up the whole screen. Tested on two devices. Linux and Windows, both hardened security and default security. This problem doesn't exist on Chromium though. Here is a screenshot showing the poor scaling https://files.catbox.moe/btm74y.png

**Comment by NickM-27** [comment]
> I'm running the latest beta2 docker image. Much better and much less tedious deleting events to clean up space if needed.
> 
> On Firefox though the delete icon is massive and takes up the whole screen. Tested on two devices. Linux and Windows, both hardened security and default security. This problem doesn't exist on Chromium though. Here is a screenshot showing the poor scaling https://files.catbox.moe/btm74y.png

PR is already in to fix this and will be in the next beta 👍

**Comment by Flaxee** [comment]
The button to delete an event in the beta2 image is working very good! 👍🏻

However, it would also be very much appreciated if there was a way to delete multiple events at the same time. For example you can use the filters & date range to find the events you want to delete, then easily mark all of them and press a delete button to delete them all at once.

**Comment by darfij** [comment]
Is there any news on this essential feature?
While most of the frigate features are comparable or advanced to other solutions, the recordings and events sorting, filtering, deleting and general manipulating are very basic.
I would even venture saying the almost complete lack of them is a show stopper!!

**Comment by NickM-27** [comment]
> Is there any news on this essential feature?
> 
> While most of the frigate features are comparable or advanced to other solutions, the recordings and events sorting, filtering, deleting and general manipulating are very basic.
> 
> I would even venture saying the almost complete lack of them is a show stopper!!

As was mentioned above have you looked at the 0.11 RC?

**Comment by darfij** [comment]
RC seems a improvement, when can we hope for a main?

**Comment by NickM-27** [comment]
> RC seems a improvement, when can we hope for a main?

What do you mean by main?

**Comment by darfij** [comment]
Sorry, master?

**Comment by NickM-27** [comment]
> Sorry, master?

Oh I see, we're looking into a last couple bugs with ffmpeg but should be good once those are figured out

**Comment by darfij** [comment]
also any chance of getting the multiple and/or sorted, and/or batch and/or all delete of events.
Single button delete is twice better than the previous two button but if you have 1000s of events?

**Comment by martymoose** [comment]
World be great delete all button or delete day

**Comment by knudsvik** [comment]
Work-around: delete the frigate.db file and restart frigate. it will build a new empty database. You still need to remove all files from the media folder.

**Comment by hmjgriffon1** [comment]
It would also be nice to be able to delete recordings, unless I am missing something there is a trash can for events but it leaves the corresponding recording behind?

**Comment by NickM-27** [comment]
> unless I am missing something there is a trash can for events but it leaves the corresponding recording behind?

recordings are deleted the next time that the recordings cleanup process runs (which is once an hour by default)

**Comment by hmjgriffon1** [comment]
> > unless I am missing something there is a trash can for events but it leaves the corresponding recording behind?
> 
> recordings are deleted the next time that the recordings cleanup process runs (which is once an hour by default)

Good to know, thanks for the quick response, love Frigate, keep up the awesome work!

**Comment by klonwar** [comment]
With basic skills of JS you can delete them like this:

1. Request all events  
```
const events = await (await fetch("http:// {ip address of your server} :5000/api/events?cameras=all&labels=all&zones=all&sub_labels=all&favorites=0&include_thumbnails=0&limit=100000", {
  "headers": {
    "accept": "application/json, text/plain, */*",
    "accept-language": "ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7,fr;q=0.6"
  },
  "referrer": "http:// {ip address of your server} :5000/events",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": null,
  "method": "GET",
  "mode": "cors",
  "credentials": "include"
})).json()
```

2. Delete all of them
```
(async () => {

    for (const event of events) {
        await fetch("http:// {ip address of your server} :5000/api/events/" + event.id, {
          "headers": {
            "accept": "application/json, text/plain, */*",
            "accept-language": "ru-RU,ru;q=0.9,en-US;q=0.8,en;q=0.7,fr;q=0.6"
          },
          "referrer": "http:// {ip address of your server} :5000/events",
          "referrerPolicy": "strict-origin-when-cross-origin",
          "body": null,
          "method": "DELETE",
          "mode": "cors",
          "credentials": "include"
        });
    }
    
})()
```

Of course this is not a solution, just a workaround I used, when found thousands of unavailable clips in web ui after experiments with configuration :)

P.S. Fetch options are simply copied from the Network tab in devtools

**Comment by klonwar** [comment]
@jasonpstokes 

> Running in Edge Dev console, with the 'CORS unblock' extension

I simply run it in Chrome console on events page without any additional extension. Have you tried disabling it?
To avoid CORS errors the origin of the site you visiting and the fetch url origin should be the same. So, just open your frigate (10.0.1.15:5000 I suppose), open console on that page and run scripts in it (with 10.0.1.15:5000 url). Just tested, seems working

![image](https://github.com/blakeblackshear/frigate/assets/12374932/7382f09d-15d9-488c-a421-725b6bb125b1)

**Comment by sochartgit** [comment]
Bump, I  am looking for a solution please.

**Comment by fuomag9** [comment]
![image](https://github.com/blakeblackshear/frigate/assets/1580624/615d31e2-a8fd-4b06-be8c-1b7ad4010ef1)

CSRF token header was missing (something changed with the latest update)

You need to add `"X-Csrf-Token":"1"` inside the `headers` section and it will work @jasonpstokes

**Comment by chamilad** [comment]
Can I suggest another approach in the way of a button to delete all events except the "saved" ones? This would enable a user workflow where I could go through the events of the day, "save" the ones I want to retain for future reference, and click a single button to clear out the rest of it? 

IMO this workflow would cover both the multi-select requests and the above suggested "Delete All" button with filters involved. In both cases, the intention is to keep a set of events and delete the rest. The "save" feature along with a single button should cover all those cases.

I started looking in to the source, at first looking at `routes/Events.jsx`. Looks like I should be able to loop through `eventPages` map and run the `onDelete()` function on each event that has `false` for `retain_indefinitely`,  when such a "Delete All" button is clicked. At least this is my idea. But I have zero idea about Preact so let me spend sometime before a PoC.

**Comment by chamilad** [comment]
I added a PR above with a PoC of what I described above. I'll upload a video of how it is from the user's point of view soon.

**Comment by NickM-27** [comment]
The frigate UI is currently undergoing a full rewrite for the next version (0.14). As part of that rewrite, there is a new "reviewed" mechanism where items are marked as reviewed and then won't be shown the next time the UI is opened unless the "show reviewed" filter is activated. That would seem to cover many of the use cases discussed here where users don't want to see things that they already reviewed.

Other users requesting this because they don't have retention working as they think it should are encouraged to create a support discussion so we can get further into why that is happening on a case by case basis.

**Comment by fuomag9** [comment]
> The frigate UI is currently undergoing a full rewrite for the next version (0.14). As part of that rewrite, there is a new "reviewed" mechanism where items are marked as reviewed and then won't be shown the next time the UI is opened unless the "show reviewed" filter is activated. That would seem to cover many of the use cases discussed here where users don't want to see things that they already reviewed.
> 
> Other users requesting this because they don't have retention working as they think it should are encouraged to create a support discussion so we can get further into why that is happening on a case by case basis.

I don't believe this should be closed. I need the OPPOSITE of retention, I need to autodelete stuff + manual mass delete of videos!

**Comment by NickM-27** [comment]
"auto delete stuff" is exactly what retention means. Setting a period to retain videos and after that they are automatically removed.

**Comment by NickM-27** [comment]
That retention period can even be minutes by the way https://docs.frigate.video/configuration/record#configuring-recording-retention

**Comment by fuomag9** [comment]
> That retention period can even be minutes by the way https://docs.frigate.video/configuration/record#configuring-recording-retention

perfect, thanks!

**Comment by NickM-27** [comment]
There is multi select to mark as reviewed or delete. However if you have that many false positives you may be better off using motion review vs object detection review in the new UI. Or work on filters or using a better model.

**Comment by darfij** [comment]
> > there is a new "reviewed" mechanism where items are marked as reviewed and then won't be shown the next time the UI is opened
> 
> Thanks @NickM-27, looking forward to the new UI!
> 
> Will we be able to multiple-select and delete events as well, please? (Or filter and select-all)
> 
> Retention works well, but I get [hundreds of false-positives](https://github.com/blakeblackshear/frigate/issues/2063#issuecomment-1828684134) every week in a warehouse, that fill the disk. I use the dev-code options posted above, but it'd be nice to have something in the UI. :)

+++ this
There are various reasons for a lot of positives that may or may not be false.
(large retention time period, a lot of people where only one/some is a positive, a animal that wonders in, .....)
a button that deletes a whole category (based on given filters like period or camera, or event type, or object type or...)
I know one should implement good practices to reduce false positives but quite often you simply have to delete a bunch of stuf.
Consider for example how it would be difficult  for you if a file explorer of your choice only had a one by one delete option!!

I would really love it if there was a UI button!!

**Comment by bmatthewshea** [comment]
It may be being designed this way, but in additional to single delete button on each event we have now, maybe also a select/deselect box.

At the top allow the current filter ('today', 'all', etc) to also be used with any bulk operation such as delete (with an extra warning maybe). You could also singly remove a 'check' if there were any you wanted to save.

That way you could select anywhere from a 'day' to 'all' and delete accordingly. Seems that would make everyone happy here - whether you want to delete all or a days worth/etc. Thanks for your work.

**Comment by neagix** [comment]
> There is multi select to mark as reviewed or delete.

I cannot find any visual clue on how to do any multi-select or delete. The scriptlet to delete all events unfortunately doesn't work due to a 401 Unauthorized error. Going through 182 events one by one is not fun.

**Comment by NickM-27** [comment]
Just long press on mobile, or on desktop you can ctrl + click, right click, or do ctrl + a to select all on the entire page
